gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequest.java

```diff
@@ -0,0 +1,331 @@
+     * Sets a collection of {@link DnsRecord}s which are to be deleted from the zone upon executing
+     * this {@code ChangeRequest}.
+     */
+    public Builder deletions(List<DnsRecord> deletions) {
+      this.deletions = Lists.newLinkedList(checkNotNull(deletions));
+      return this;
+    }
+
+    /**
+     * Associates a server-assigned id to this {@code ChangeRequest}.
+     */
+    Builder id(String id) {
+      this.id = checkNotNull(id);
+      return this;
+    }
+
+    /**
+     * Sets the time when this {@code ChangeRequest} was started by a server.
+     */
+    Builder startTimeMillis(long startTimeMillis) {
+      this.startTimeMillis = startTimeMillis;
+      return this;
+    }
+
+    /**
+     * Sets the current status of this {@code ChangeRequest}.
+     */
+    Builder status(Status status) {
+      this.status = checkNotNull(status);
+      return this;
+    }
+
+    /**
+     * Creates a {@code ChangeRequest} instance populated by the values associated with this
+     * builder.
+     */
+    public ChangeRequest build() {
+      return new ChangeRequest(this);
+    }
+  }
+
+  private ChangeRequest(Builder builder) {
+    this.additions = builder.additions;
+    this.deletions = builder.deletions;
+    this.id = builder.id;
+    this.startTimeMillis = builder.startTimeMillis;
+    this.status = builder.status;
+  }
+
+  /**
+   * Returns an empty builder for the {@code ChangeRequest} class.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Creates a builder populated with values of this {@code ChangeRequest}.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Returns the list of {@link DnsRecord}s to be added to the zone upon executing this {@code
+   * ChangeRequest}.
+   */
+  public List<DnsRecord> additions() {
+    return additions;
+  }
+
+  /**
+   * Returns the list of {@link DnsRecord}s to be deleted from the zone upon executing this {@code
+   * ChangeRequest}.
+   */
+  public List<DnsRecord> deletions() {
+    return deletions;
+  }
+
+  /**
+   * Returns the id assigned to this {@code ChangeRequest} by the server.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the time when this {@code ChangeRequest} was started by the server.
+   */
+  public Long startTimeMillis() {
+    return startTimeMillis;
+  }
+
+  /**
+   * Returns the status of this {@code ChangeRequest}.
+   */
+  public Status status() {
+    return status;
+  }
+
+  com.google.api.services.dns.model.Change toPb() {
+    com.google.api.services.dns.model.Change pb =
+        new com.google.api.services.dns.model.Change();
+    // set id
+    if (id() != null) {
+      pb.setId(id());
+    }
+    // set timestamp
+    if (startTimeMillis() != null) {
+      pb.setStartTime(ISODateTimeFormat.dateTime().withZoneUTC().print(startTimeMillis()));
+    }
+    // set status
+    if (status() != null) {
+      pb.setStatus(status().status);
```

Comment:
lets remove `status` private field. and make this one calls `toLowerCase` to be consistent with `fromPb`